bundles/com.salesforce.bazel.eclipse.core/src/com/salesforce/bazel/eclipse/core/model/discovery/BaseProvisioningStrategy.java

```diff
@@ -123,14 +125,61 @@ public abstract class BaseProvisioningStrategy implements TargetProvisioningStra
 
     private static Logger LOG = LoggerFactory.getLogger(BaseProvisioningStrategy.class);
 
-    private BazelProjectFileSystemMapper fileSystemMapper;
+    /**
+     * Creates a problem marker of type {@link BazelCoreSharedContstants#CLASSPATH_CONTAINER_PROBLEM_MARKER} for the
+     * given status.
+     *
+     * @param project
+     *            the project to create the marker at (must not be <code>null</code>)
+     * @param status
+     *            the status to create the marker for (must not be <code>null</code>)
+     * @return the created marker (never <code>null</code>)
+     * @throws CoreException
+     */
+    static IMarker createClasspathContainerProblem(BazelProject project, IStatus status) throws CoreException {
```

Comment:
I moved the validation logic into JavaAspectClasspathInfo, so marked this package protected, static. Happy to move this out into a helper class, let me know